### ISSUE TITLE
Paragraphs are unnecessarily merged

### ISSUE THREAD

**Issue by jazblueful** [ianstormtaylor/slate#3466]
#### Do you want to request a _feature_ or report a _bug_?

Bug

#### What's the current behavior?
https://codesandbox.io/s/slate-reproductions-x6nf8

Put cursor inside bold text
Tap backspace until you're at 2
Press backspace again
Number 2 is deleted and both paragraphs are merged into one

Slate: 0.57.1
Browser: Chrome
OS: Linux

#### What's the expected behavior?
Number 2 is deleted without merging the paragraphs into one

**Comment by dimitriadamou** [comment]
Does this happen on the 0.58 version? https://codesandbox.io/s/slate-reproductions-2yr90?file=/index.js

Testing from Mac on Chrome, no replicable issue
![May-06-2020 19-24-31](https://user-images.githubusercontent.com/3094239/81161651-3e08fc80-8fcf-11ea-99e5-15bfbe67bbb4.gif)

**Comment by jazblueful** [comment]
Yes, the bug is still there. I just tested it on both Windows and Linux version of Chrome. Btw you need to put the cursor in the MIDDLE of the bold text before tapping backspace in order to replicate the bug.

**Comment by dimitriadamou** [comment]
Ah I see now, if I get time over the weekend I may have a play with it

**Comment by udayanshevade** [comment]
Looks like a duplicate of #3339

I took a crack at fixing it:

![richtext_char_backspace_fix_2](https://user-images.githubusercontent.com/11330558/93726641-065cb480-fb6c-11ea-9ff5-a7698c0a1a4d.gif)
